src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/HashBuilder.cs

```diff
@@ -28,25 +45,29 @@ public static string BuildIDFromSymbol(ITypeSymbol typeSymbol)
 		/// <param name="input">The string to hash</param>
 		/// <param name="algoritm">The algorithm to use</param>
 		/// <returns>A hex-formatted string of the hash</returns>
-		public static string Build(string input, HashAlgorithm algoritm = null)
+		public static string Build(string input)
 		{
-			using (algoritm ??= SHA256.Create())
+			using (var algoritm = SHA256.Create())
 			{
 				// Convert the input string to a byte array and compute the hash.
 				var data = algoritm.ComputeHash(Encoding.UTF8.GetBytes(input));
 
-				// Create a new Stringbuilder to collect the bytes
-				// and create a string.
-				var builder = new StringBuilder();
-
-				// Use the first 16 bytes of the hash
-				for (int i = 0; i < Math.Min(data.Length, 16); i++)
+				var min = Math.Min(data.Length, 16);
+				var result = new string((char)0, min * 2);
```

Comment:
if you're using `unsafe` then maybe you should also `stackalloc` with a `Span<Char>`?

Reply:
@spouliot We'll endup calling `.ToString()`, I think. So I guess creating it as string in the first place could be faster, but I'm not really sure. Haven't benchmarked.

Are there any concerns with the current approach?

Reply:
yes, the string has to be created, eventually :)
but the stackalloc is cheap, use `char` (easier logic) and like @Sergio0694 commented it avoid modifying the create string (bad)